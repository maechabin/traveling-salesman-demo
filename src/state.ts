import { State } from './state.model';

const state: State = {
  departure: {
    title: '山王パークタワー',
    lat: 35.6731652,
    lng: 139.7407961,
    label: '発',
  },
  arival: {
    title: '六本木ヒルズ森タワー',
    lat: 35.6604638,
    lng: 139.72924869999997,
    label: '着',
  },
  routes: [
    {
      id: 1,
      title: 'Wコンフォートタワーズ EAST',
      lat: 35.6471312,
      lng: 139.80113460000007,
      label: 'A',
      sortId: 0,
    },
    {
      id: 2,
      title: 'セルリアンタワー',
      lat: 35.656325,
      lng: 139.699608,
      label: 'B',
      sortId: 0,
    },
    {
      id: 3,
      title: '新宿センタービル',
      lat: 35.6917502,
      lng: 139.6955153,
      label: 'C',
      sortId: 0,
    },
    {
      id: 4,
      title: 'JPタワー',
      lat: 35.679799,
      lng: 139.764545,
      label: 'D',
      sortId: 0,
    },
    {
      id: 5,
      title: '新宿パークタワー',
      lat: 35.68564,
      lng: 139.69094999999993,
      label: 'E',
      sortId: 0,
    },
    {
      id: 6,
      title: '住友不動産新宿グランドタワー',
      lat: 35.6957954,
      lng: 139.69038920000003,
      label: 'F',
      sortId: 0,
    },
    {
      id: 7,
      title: '勝どきビュータワー',
      lat: 35.6595554,
      lng: 139.77651500000002,
      label: 'G',
      sortId: 0,
    },
    {
      id: 8,
      title: 'サンシャイン６０',
      lat: 35.7291213,
      lng: 139.7191322,
      label: 'H',
      sortId: 0,
    },
  ],
  initialFlag: true,
  choosingRouteStartFlag: false,
  choosingRouteFinishFlag: false,
  viewAnswerFlag: false,
  currentSortId: 8, // routes（経路）の要素数を指定する
  transport: 'car',
  expressway: 'no',
  traffic: 'standard',
  departureTime: new Date(Date.now() + 1000 * 60 * 60 * 24 * 3),
  gross: {
    distance: 0, // 総距離
    duration: 0, // 総時間
  },
  answerGross: {
    distance: 0, // 正解の総距離
    duration: 0, // 正解の総時間
  },
  answerWaypointOrder: [], // 正解の経路の表示順
};

export default state;
