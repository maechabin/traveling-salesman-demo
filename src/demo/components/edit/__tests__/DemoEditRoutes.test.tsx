import React from 'react';
import { shallow } from 'enzyme';

import { Route } from '../../../../state.model';
import DemoEditRoutes from '../DemoEditRoutes';

describe('<DemoDescription />', () => {
  it('snapshot', () => {
    // setup
    const routes: Route[] = [
      { id: 1, title: '秋葉原駅', lat: 111, lng: 111, label: 'A', sortId: 5 },
      { id: 2, title: '文京区役所', lat: 222, lng: 222, label: 'B', sortId: 4 },
      { id: 3, title: '明治大学', lat: 333, lng: 333, label: 'C', sortId: 1 },
      { id: 4, title: '浜離宮', lat: 444, lng: 444, label: 'D', sortId: 2 },
      { id: 5, title: '国会議事堂', lat: 555, lng: 555, label: 'E', sortId: 8 },
      { id: 6, title: '東京タワー', lat: 666, lng: 666, label: 'F', sortId: 3 },
      { id: 7, title: '上野動物園', lat: 777, lng: 777, label: 'G', sortId: 7 },
      { id: 8, title: '東京駅', lat: 888, lng: 888, label: 'H', sortId: 6 },
    ];
    const callbackSpy = jest.fn();
    const props = {
      callback: callbackSpy,
      routes,
    };

    // exercise
    const wrapper = shallow(<DemoEditRoutes {...props} />);

    // verify
    expect(wrapper).toMatchSnapshot();
  });

  it('handleChange', () => {
    // setup
    const routes: Route[] = [
      { id: 1, title: '秋葉原駅', lat: 111, lng: 111, label: 'A', sortId: 5 },
      { id: 2, title: '文京区役所', lat: 222, lng: 222, label: 'B', sortId: 4 },
      { id: 3, title: '明治大学', lat: 333, lng: 333, label: 'C', sortId: 1 },
      { id: 4, title: '浜離宮', lat: 444, lng: 444, label: 'D', sortId: 2 },
      { id: 5, title: '国会議事堂', lat: 555, lng: 555, label: 'E', sortId: 8 },
      { id: 6, title: '東京タワー', lat: 666, lng: 666, label: 'F', sortId: 3 },
      { id: 7, title: '上野動物園', lat: 777, lng: 777, label: 'G', sortId: 7 },
      { id: 8, title: '東京駅', lat: 888, lng: 888, label: 'H', sortId: 6 },
    ];
    const callbackSpy = jest.fn();
    const props = {
      callback: callbackSpy,
      routes,
    };
    const event = 'event';
    const wrapper = shallow(<DemoEditRoutes {...props} />);

    // exercise
    wrapper.find('input').forEach((input, index) => {
      input.simulate('change', event);

      // verify
      expect(callbackSpy).toHaveBeenCalledWith(event, index + 1);
    });
  });
});
