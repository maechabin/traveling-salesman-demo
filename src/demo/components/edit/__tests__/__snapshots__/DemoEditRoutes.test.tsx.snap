// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DemoDescription /> snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DemoEditRoutes
    callback={[MockFunction]}
    routes={
      Array [
        Object {
          "id": 1,
          "label": "A",
          "lat": 111,
          "lng": 111,
          "sortId": 5,
          "title": "秋葉原駅",
        },
        Object {
          "id": 2,
          "label": "B",
          "lat": 222,
          "lng": 222,
          "sortId": 4,
          "title": "文京区役所",
        },
        Object {
          "id": 3,
          "label": "C",
          "lat": 333,
          "lng": 333,
          "sortId": 1,
          "title": "明治大学",
        },
        Object {
          "id": 4,
          "label": "D",
          "lat": 444,
          "lng": 444,
          "sortId": 2,
          "title": "浜離宮",
        },
        Object {
          "id": 5,
          "label": "E",
          "lat": 555,
          "lng": 555,
          "sortId": 8,
          "title": "国会議事堂",
        },
        Object {
          "id": 6,
          "label": "F",
          "lat": 666,
          "lng": 666,
          "sortId": 3,
          "title": "東京タワー",
        },
        Object {
          "id": 7,
          "label": "G",
          "lat": 777,
          "lng": 777,
          "sortId": 7,
          "title": "上野動物園",
        },
        Object {
          "id": 8,
          "label": "H",
          "lat": 888,
          "lng": 888,
          "sortId": 6,
          "title": "東京駅",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          style={
            Object {
              "backgroundColor": "#7fbc39",
              "boxSizing": "border-box",
              "color": "#fff",
              "display": "block",
              "padding": "4px",
              "textAlign": "center",
              "width": "320px",
            }
          }
        >
          ルート
        </label>,
        Array [
          <li>
            <input
              defaultValue="秋葉原駅"
              onChange={[Function]}
              style={
                Object {
                  "boxSizing": "border-box",
                  "fontSize": "16px",
                  "imeMode": "active",
                  "padding": "6px 4px",
                  "width": "320px",
                }
              }
            />
          </li>,
          <li>
            <input
              defaultValue="文京区役所"
              onChange={[Function]}
              style={
                Object {
                  "boxSizing": "border-box",
                  "fontSize": "16px",
                  "imeMode": "active",
                  "padding": "6px 4px",
                  "width": "320px",
                }
              }
            />
          </li>,
          <li>
            <input
              defaultValue="明治大学"
              onChange={[Function]}
              style={
                Object {
                  "boxSizing": "border-box",
                  "fontSize": "16px",
                  "imeMode": "active",
                  "padding": "6px 4px",
                  "width": "320px",
                }
              }
            />
          </li>,
          <li>
            <input
              defaultValue="浜離宮"
              onChange={[Function]}
              style={
                Object {
                  "boxSizing": "border-box",
                  "fontSize": "16px",
                  "imeMode": "active",
                  "padding": "6px 4px",
                  "width": "320px",
                }
              }
            />
          </li>,
          <li>
            <input
              defaultValue="国会議事堂"
              onChange={[Function]}
              style={
                Object {
                  "boxSizing": "border-box",
                  "fontSize": "16px",
                  "imeMode": "active",
                  "padding": "6px 4px",
                  "width": "320px",
                }
              }
            />
          </li>,
          <li>
            <input
              defaultValue="東京タワー"
              onChange={[Function]}
              style={
                Object {
                  "boxSizing": "border-box",
                  "fontSize": "16px",
                  "imeMode": "active",
                  "padding": "6px 4px",
                  "width": "320px",
                }
              }
            />
          </li>,
          <li>
            <input
              defaultValue="上野動物園"
              onChange={[Function]}
              style={
                Object {
                  "boxSizing": "border-box",
                  "fontSize": "16px",
                  "imeMode": "active",
                  "padding": "6px 4px",
                  "width": "320px",
                }
              }
            />
          </li>,
          <li>
            <input
              defaultValue="東京駅"
              onChange={[Function]}
              style={
                Object {
                  "boxSizing": "border-box",
                  "fontSize": "16px",
                  "imeMode": "active",
                  "padding": "6px 4px",
                  "width": "320px",
                }
              }
            />
          </li>,
        ],
      ],
      "style": Object {
        "listStyleType": "none",
        "margin": "0 0 16px 0",
        "padding": 0,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ルート",
          "style": Object {
            "backgroundColor": "#7fbc39",
            "boxSizing": "border-box",
            "color": "#fff",
            "display": "block",
            "padding": "4px",
            "textAlign": "center",
            "width": "320px",
          },
        },
        "ref": null,
        "rendered": "ルート",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": <input
            defaultValue="秋葉原駅"
            onChange={[Function]}
            style={
              Object {
                "boxSizing": "border-box",
                "fontSize": "16px",
                "imeMode": "active",
                "padding": "6px 4px",
                "width": "320px",
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "defaultValue": "秋葉原駅",
            "onChange": [Function],
            "style": Object {
              "boxSizing": "border-box",
              "fontSize": "16px",
              "imeMode": "active",
              "padding": "6px 4px",
              "width": "320px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <input
            defaultValue="文京区役所"
            onChange={[Function]}
            style={
              Object {
                "boxSizing": "border-box",
                "fontSize": "16px",
                "imeMode": "active",
                "padding": "6px 4px",
                "width": "320px",
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "defaultValue": "文京区役所",
            "onChange": [Function],
            "style": Object {
              "boxSizing": "border-box",
              "fontSize": "16px",
              "imeMode": "active",
              "padding": "6px 4px",
              "width": "320px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": <input
            defaultValue="明治大学"
            onChange={[Function]}
            style={
              Object {
                "boxSizing": "border-box",
                "fontSize": "16px",
                "imeMode": "active",
                "padding": "6px 4px",
                "width": "320px",
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "defaultValue": "明治大学",
            "onChange": [Function],
            "style": Object {
              "boxSizing": "border-box",
              "fontSize": "16px",
              "imeMode": "active",
              "padding": "6px 4px",
              "width": "320px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": <input
            defaultValue="浜離宮"
            onChange={[Function]}
            style={
              Object {
                "boxSizing": "border-box",
                "fontSize": "16px",
                "imeMode": "active",
                "padding": "6px 4px",
                "width": "320px",
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "defaultValue": "浜離宮",
            "onChange": [Function],
            "style": Object {
              "boxSizing": "border-box",
              "fontSize": "16px",
              "imeMode": "active",
              "padding": "6px 4px",
              "width": "320px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "host",
        "props": Object {
          "children": <input
            defaultValue="国会議事堂"
            onChange={[Function]}
            style={
              Object {
                "boxSizing": "border-box",
                "fontSize": "16px",
                "imeMode": "active",
                "padding": "6px 4px",
                "width": "320px",
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "defaultValue": "国会議事堂",
            "onChange": [Function],
            "style": Object {
              "boxSizing": "border-box",
              "fontSize": "16px",
              "imeMode": "active",
              "padding": "6px 4px",
              "width": "320px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "host",
        "props": Object {
          "children": <input
            defaultValue="東京タワー"
            onChange={[Function]}
            style={
              Object {
                "boxSizing": "border-box",
                "fontSize": "16px",
                "imeMode": "active",
                "padding": "6px 4px",
                "width": "320px",
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "defaultValue": "東京タワー",
            "onChange": [Function],
            "style": Object {
              "boxSizing": "border-box",
              "fontSize": "16px",
              "imeMode": "active",
              "padding": "6px 4px",
              "width": "320px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "6",
        "nodeType": "host",
        "props": Object {
          "children": <input
            defaultValue="上野動物園"
            onChange={[Function]}
            style={
              Object {
                "boxSizing": "border-box",
                "fontSize": "16px",
                "imeMode": "active",
                "padding": "6px 4px",
                "width": "320px",
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "defaultValue": "上野動物園",
            "onChange": [Function],
            "style": Object {
              "boxSizing": "border-box",
              "fontSize": "16px",
              "imeMode": "active",
              "padding": "6px 4px",
              "width": "320px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "7",
        "nodeType": "host",
        "props": Object {
          "children": <input
            defaultValue="東京駅"
            onChange={[Function]}
            style={
              Object {
                "boxSizing": "border-box",
                "fontSize": "16px",
                "imeMode": "active",
                "padding": "6px 4px",
                "width": "320px",
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "defaultValue": "東京駅",
            "onChange": [Function],
            "style": Object {
              "boxSizing": "border-box",
              "fontSize": "16px",
              "imeMode": "active",
              "padding": "6px 4px",
              "width": "320px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            style={
              Object {
                "backgroundColor": "#7fbc39",
                "boxSizing": "border-box",
                "color": "#fff",
                "display": "block",
                "padding": "4px",
                "textAlign": "center",
                "width": "320px",
              }
            }
          >
            ルート
          </label>,
          Array [
            <li>
              <input
                defaultValue="秋葉原駅"
                onChange={[Function]}
                style={
                  Object {
                    "boxSizing": "border-box",
                    "fontSize": "16px",
                    "imeMode": "active",
                    "padding": "6px 4px",
                    "width": "320px",
                  }
                }
              />
            </li>,
            <li>
              <input
                defaultValue="文京区役所"
                onChange={[Function]}
                style={
                  Object {
                    "boxSizing": "border-box",
                    "fontSize": "16px",
                    "imeMode": "active",
                    "padding": "6px 4px",
                    "width": "320px",
                  }
                }
              />
            </li>,
            <li>
              <input
                defaultValue="明治大学"
                onChange={[Function]}
                style={
                  Object {
                    "boxSizing": "border-box",
                    "fontSize": "16px",
                    "imeMode": "active",
                    "padding": "6px 4px",
                    "width": "320px",
                  }
                }
              />
            </li>,
            <li>
              <input
                defaultValue="浜離宮"
                onChange={[Function]}
                style={
                  Object {
                    "boxSizing": "border-box",
                    "fontSize": "16px",
                    "imeMode": "active",
                    "padding": "6px 4px",
                    "width": "320px",
                  }
                }
              />
            </li>,
            <li>
              <input
                defaultValue="国会議事堂"
                onChange={[Function]}
                style={
                  Object {
                    "boxSizing": "border-box",
                    "fontSize": "16px",
                    "imeMode": "active",
                    "padding": "6px 4px",
                    "width": "320px",
                  }
                }
              />
            </li>,
            <li>
              <input
                defaultValue="東京タワー"
                onChange={[Function]}
                style={
                  Object {
                    "boxSizing": "border-box",
                    "fontSize": "16px",
                    "imeMode": "active",
                    "padding": "6px 4px",
                    "width": "320px",
                  }
                }
              />
            </li>,
            <li>
              <input
                defaultValue="上野動物園"
                onChange={[Function]}
                style={
                  Object {
                    "boxSizing": "border-box",
                    "fontSize": "16px",
                    "imeMode": "active",
                    "padding": "6px 4px",
                    "width": "320px",
                  }
                }
              />
            </li>,
            <li>
              <input
                defaultValue="東京駅"
                onChange={[Function]}
                style={
                  Object {
                    "boxSizing": "border-box",
                    "fontSize": "16px",
                    "imeMode": "active",
                    "padding": "6px 4px",
                    "width": "320px",
                  }
                }
              />
            </li>,
          ],
        ],
        "style": Object {
          "listStyleType": "none",
          "margin": "0 0 16px 0",
          "padding": 0,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ルート",
            "style": Object {
              "backgroundColor": "#7fbc39",
              "boxSizing": "border-box",
              "color": "#fff",
              "display": "block",
              "padding": "4px",
              "textAlign": "center",
              "width": "320px",
            },
          },
          "ref": null,
          "rendered": "ルート",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <input
              defaultValue="秋葉原駅"
              onChange={[Function]}
              style={
                Object {
                  "boxSizing": "border-box",
                  "fontSize": "16px",
                  "imeMode": "active",
                  "padding": "6px 4px",
                  "width": "320px",
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "defaultValue": "秋葉原駅",
              "onChange": [Function],
              "style": Object {
                "boxSizing": "border-box",
                "fontSize": "16px",
                "imeMode": "active",
                "padding": "6px 4px",
                "width": "320px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <input
              defaultValue="文京区役所"
              onChange={[Function]}
              style={
                Object {
                  "boxSizing": "border-box",
                  "fontSize": "16px",
                  "imeMode": "active",
                  "padding": "6px 4px",
                  "width": "320px",
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "defaultValue": "文京区役所",
              "onChange": [Function],
              "style": Object {
                "boxSizing": "border-box",
                "fontSize": "16px",
                "imeMode": "active",
                "padding": "6px 4px",
                "width": "320px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <input
              defaultValue="明治大学"
              onChange={[Function]}
              style={
                Object {
                  "boxSizing": "border-box",
                  "fontSize": "16px",
                  "imeMode": "active",
                  "padding": "6px 4px",
                  "width": "320px",
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "defaultValue": "明治大学",
              "onChange": [Function],
              "style": Object {
                "boxSizing": "border-box",
                "fontSize": "16px",
                "imeMode": "active",
                "padding": "6px 4px",
                "width": "320px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": <input
              defaultValue="浜離宮"
              onChange={[Function]}
              style={
                Object {
                  "boxSizing": "border-box",
                  "fontSize": "16px",
                  "imeMode": "active",
                  "padding": "6px 4px",
                  "width": "320px",
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "defaultValue": "浜離宮",
              "onChange": [Function],
              "style": Object {
                "boxSizing": "border-box",
                "fontSize": "16px",
                "imeMode": "active",
                "padding": "6px 4px",
                "width": "320px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": <input
              defaultValue="国会議事堂"
              onChange={[Function]}
              style={
                Object {
                  "boxSizing": "border-box",
                  "fontSize": "16px",
                  "imeMode": "active",
                  "padding": "6px 4px",
                  "width": "320px",
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "defaultValue": "国会議事堂",
              "onChange": [Function],
              "style": Object {
                "boxSizing": "border-box",
                "fontSize": "16px",
                "imeMode": "active",
                "padding": "6px 4px",
                "width": "320px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "host",
          "props": Object {
            "children": <input
              defaultValue="東京タワー"
              onChange={[Function]}
              style={
                Object {
                  "boxSizing": "border-box",
                  "fontSize": "16px",
                  "imeMode": "active",
                  "padding": "6px 4px",
                  "width": "320px",
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "defaultValue": "東京タワー",
              "onChange": [Function],
              "style": Object {
                "boxSizing": "border-box",
                "fontSize": "16px",
                "imeMode": "active",
                "padding": "6px 4px",
                "width": "320px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "host",
          "props": Object {
            "children": <input
              defaultValue="上野動物園"
              onChange={[Function]}
              style={
                Object {
                  "boxSizing": "border-box",
                  "fontSize": "16px",
                  "imeMode": "active",
                  "padding": "6px 4px",
                  "width": "320px",
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "defaultValue": "上野動物園",
              "onChange": [Function],
              "style": Object {
                "boxSizing": "border-box",
                "fontSize": "16px",
                "imeMode": "active",
                "padding": "6px 4px",
                "width": "320px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "7",
          "nodeType": "host",
          "props": Object {
            "children": <input
              defaultValue="東京駅"
              onChange={[Function]}
              style={
                Object {
                  "boxSizing": "border-box",
                  "fontSize": "16px",
                  "imeMode": "active",
                  "padding": "6px 4px",
                  "width": "320px",
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "defaultValue": "東京駅",
              "onChange": [Function],
              "style": Object {
                "boxSizing": "border-box",
                "fontSize": "16px",
                "imeMode": "active",
                "padding": "6px 4px",
                "width": "320px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
